tests/test262_runner.rs

```diff
@@ -0,0 +1,589 @@
+use clap::Parser as ClapParser;
+use serde::{Deserialize, Serialize};
+use std::{
+    borrow::Cow,
+    collections::HashMap,
+    ffi::OsStr,
+    fs::{read_dir, File},
+    io::{ErrorKind, Read},
+    path::PathBuf,
+    process::{Command, Stdio},
+    time::Duration,
+};
+use wait_timeout::ChildExt;
+
+#[derive(Deserialize, Serialize, Debug, Clone, Copy, PartialEq, Eq)]
+#[serde(rename_all = "UPPERCASE")]
+enum TestExpectation {
+    Pass,
+    Fail,
+    Crash,
+    Timeout,
+}
+
+/// These directory names are filtered out by default.
+const SKIP_DIRS: &[&str] = &[
+    "annexB", "intl402", "staging",
+    //
+    //"built-ins",
+    //"language",
+];
+
+fn is_test_file(file_name: &str) -> bool {
+    file_name.ends_with(".js") && !file_name.contains("_FIXTURE")
+}
+
+#[derive(Debug)]
+struct Test262Runner {
+    runner_base_path: PathBuf,
+    tests_base: PathBuf,
+    nova_harness_path: PathBuf,
+    nova_cli_path: PathBuf,
+    expectations: HashMap<PathBuf, TestExpectation>,
+    treat_crashes_as_failures: bool,
+
+    num_tests_run: usize,
```

Comment:
**suggestion:** Should we maybe split the above configuration parts into their own struct and keep these mutable parts in a separate part? There might be some magical benefit to that :)

Reply:
In a follow-up I will make the test runner parallel, and there these fields will live in a mutable thread-local struct, which then gets joined. But for now, while the test runner is serial, I think we can keep this as a temporary unoptimal solution.